src/main/java/com/ood/myorange/controller/StorageConfigController.java

```diff
@@ -0,0 +1,99 @@
+package com.ood.myorange.controller;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.ood.myorange.dto.StorageConfigDto;
+import com.ood.myorange.exception.InvalidRequestException;
+import com.ood.myorange.service.StorageConfigService;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+/**
+ * Created by Guancheng Lai on 3/19/2020
+ */
+
+@RestController
+@Slf4j
+@RequestMapping("/admin")
+public class StorageConfigController {
+
+    @Autowired
+    StorageConfigService storageConfigService;
+
+    @GetMapping("/config/storage")
+    public List<StorageConfigDto> viewAllStorageConfig() throws JsonProcessingException {
+        return storageConfigService.getAllConfigurations();
+    }
+
+    @GetMapping("/config/storage/{configId}")
+    public StorageConfigDto viewOneStorageConfig(
+            @PathVariable("configId") int configId
+    ) throws JsonProcessingException {
+        validateConfigId( configId );
+        return storageConfigService.getConfiguration( configId );
+    }
+
+    @PostMapping("/config/storage")
+    public void addStorageConfig(
+            @RequestBody StorageConfigDto storageConfigDto
+    ) throws JsonProcessingException {
+        validateConfig( storageConfigDto );
+        storageConfigService.addConfiguration( storageConfigDto );
+    }
+
+    @PutMapping("/config/storage")
+    public void editStorageConfig(
+            @RequestBody StorageConfigDto storageConfigDto
+    ) throws JsonProcessingException {
+        validateConfig( storageConfigDto );
+        storageConfigService.updateConfiguration(storageConfigDto );
+    }
+
+    @DeleteMapping("/config/storage/{configId}")
+    public void deleteStorageConfig(
+            @PathVariable int configId
+    ){
+        validateConfigId( configId );
+        storageConfigService.deleteConfiguration( configId );
+    }
+
+    void validateConfigId(int configId) {
+        if (configId <= 0) {
+            throw new InvalidRequestException("ConfigId should be greater than zero");
+        }
+    }
+
+    void validateConfig(StorageConfigDto storageConfigDto) {
+        switch (storageConfigDto.getType()) {
+            case "AWS" :
+                if (
+                        storageConfigDto.getAwsAccessKeyId() == null
+                                || storageConfigDto.getAwsSecretAccessKey() == null
+                                || storageConfigDto.getAwsRegion() == null
+                                || storageConfigDto.getAwsBucketName() == null
+                ) {
+                    throw new InvalidRequestException( "Incomplete AWS S3 configuration"
```

Comment:
If you want to change, you can try to put the data into a map and append it in the second parameter.
like ''throw new InvalidRequestException( "Incomplete AWS S3 configuration", map);"
If you don't, this format is also ok.

Reply:
Probably I am not going to change it. It is only easier to read.